packages/e2e/test/wallet/util.ts

```diff
@@ -22,6 +24,16 @@ export const firstValueFromTimed = <T>(
 export const waitForWalletStateSettle = (wallet: ObservableWallet) =>
   firstValueFromTimed(
     wallet.syncStatus.isSettled$.pipe(filter((isSettled) => isSettled)),
-    'Took too long to load',
+    'Took too long to settle',
     SYNC_TIMEOUT
   );
+
+export const waitForWalletBalance = (wallet: ObservableWallet) =>
+  firstValueFromTimed(
+    wallet.balance.utxo.total$.pipe(filter(({ coins }) => coins > 0)),
+    'Took too long to load balance',
+    BALANCE_TIMEOUT
+  );
+
+export const walletReady = (wallet: ObservableWallet) =>
+  Promise.all([waitForWalletStateSettle(wallet), waitForWalletBalance(wallet)]);
```

Comment:
Not sure if this is an issue that would surface, but the following scenario is possible:
1. Wallet settles with 0 balance (promise resolves)
2. Wallet fetches balance >0 (promise resolves), but wallet is still fetching other data that's important for the test

```suggestion
  firstValueFrom(combineLatest([state$, balance$]).pipe(filter(([state, balance]) => state === settled && balance.coins > 0n)));
```

Reply:
addressed here: https://github.com/input-output-hk/cardano-js-sdk/pull/435/commits/25d6f7a4a630e4b456ca94930b5a44468d6368c8